src/npc.cpp

```diff
@@ -624,10 +624,144 @@ void starting_inv( npc &who, const npc_class_id &type )
     who.inv += res;
 }
 
+void npc_log_manager::display( std::vector<int> filter_ids )
+{
+    std::vector<std::string> res;
+    std::vector<npc_work_log_entry> log_vec;
+    if( filter_ids.empty() ) {
+        log_vec = get_log_as_vector();
+    } else if( static_cast<int>( filter_ids.size() ) == 1 ) {
+        log_vec = get_log_filtered_by_id( filter_ids[0] );
+    } else {
+        log_vec = get_log_filtered_by_ids( filter_ids );
+    }
+    for( const npc_work_log_entry entry : log_vec ) {
+        // work log does double duty as a debug log of sorts.
+        // and also feedback as to why NPCs fail to do activities etc.
+        if( debug_mode ) {
+            res.push_back( entry.entry_to_string() );
+        } else {
+            if( entry.get_entry_type() == ENTRY_WORK_RESULT ||
+                entry.get_entry_type() == ENTRY_CHANGED_ACTIVITY ) {
+                res.push_back( entry.entry_to_string() );
+            }
+        }
+    }
+    display( res );
+}
+
+void npc_log_manager::display( character_id filter_id )
+{
+    std::vector<int> res;
+    res.push_back( filter_id.get_value() );
+    display( res );
+}
+
+void npc_log_manager::display( std::vector<std::string> entries )
+{
+    if( entries.empty() ) {
+        return;
+    }
+    folded.clear();
+    w_width = std::min( TERMX, FULL_SCREEN_WIDTH );
+    w_height = std::min( TERMY, FULL_SCREEN_HEIGHT );
+    w_x = ( TERMX - w_width ) / 2;
+    w_y = ( TERMY - w_height ) / 2;
+
+    w = catacurses::newwin( w_height, w_width, point( w_x, w_y ) );
+
+    ctxt = input_context();
+    ctxt.register_action( "UP" );
+    ctxt.register_action( "DOWN" );
+    ctxt.register_action( "QUIT" );
+    ctxt.register_action( "PAGE_UP" );
+    ctxt.register_action( "PAGE_DOWN" );
+
```

Comment:
```suggestion
    ctxt.register_action( "HELP_KEYBINDINGS" );

```
So player can change the keybindings.